code/modules/projectiles/projectile.dm

```diff
@@ -302,6 +302,21 @@
 				Range()
 			sleep(1)
 
+obj/item/projectile/proc/reflect_back(atom/source, list/position_modifiers = list(0, 0, 0, 0, 0, -1, 1, -2, 2))
+	if(starting)
+		var/new_x = starting.x + pick(position_modifiers)
+		var/new_y = starting.y + pick(position_modifiers)
+		var/turf/curloc = get_turf(source)
+
+		// redirect the projectile
+		firer = source

```

Comment:
This means that you can only use this proc with a mob, never with a structure or anything else without causing runtimes

Reply:
Good catch. Didn't know what the vars meant in all honesty. Changed in the coming commit